prometheus-port.cabal

```diff
@@ -26,15 +26,43 @@ library
 
   hs-source-dirs: src
 
-  ghc-options: -O1
```

Comment:
You forgot to put this one back in.

Reply:
No, libraries should not use `-O1`, only binaries.

Reply:
I have not seeing anything supporting this claim, merely the line in haskell/aeson#846 reciting the `cabal check` string that explicitly talks about `-O2`. When I compile modules it does not recompile every single underlying library based on the optimization level and `cabal check` never mentions this.

Reply:
Have a look at hackage. Libraries don't use any optimization flags. Binaries are the one that transitively compile all dependencies with the optimization flags. 

A trivial example of this is `coinweb-server`. @jproyo fixed our build to never compile with any `-OX` flag on local to avoid this problem. Otherwise, we had to recompile the project multiple times (and all its dependencies) when we build the library or the tests.

Reply:
> Libraries don't use any optimization flags

`bytestring`/`text`/`binary`/ `cereal` use `-O2`. You're correct in that it's libraries don't seem to use `-O1`, but this doesn't prove your point outright.

Reply:
You may be right. It is surprising amount of libraries that do not use `-ON` and some of them are performant ones .

Reply:
I will repeat the benchmarks with `-O2`.